### PR TITLE
Make editmodeImagePreview available for YouTube videos

### PR DESCRIPTION
**Pull request by bitfactory-jurgen-jansma**
## Changes in this pull request  
The config `editmodeImagePreview` for the video editable was not working for YouTube videos. I made it working in this pull request. 

## Additional info  
I'm not sure if you see this as a bug fix or as a feature/improvement. But, in our case, this solves a bug in loading the editor in the admin panel, so that's why I submitted this pull request in 10.5.

### CODE REVIEW

**Comment by brusch** [comment]
@bitfactory-jurgen-jansma thanks a lot for your PR! 
2 thinks we should consider as well: 
1. We should also use `$inAdmin` argument for `getYoutubeCode()` same as for `getAssetCode()`, see: https://github.com/pimcore/pimcore/blob/b447bc02341a2eb097bc48c2f6e9f3f84d708829/models/Document/Editable/Video.php#L457
2. to be consisten we should try to fix that for Vimeo and Dailymotion as well

Thanks a lot!

**Comment by bitfactory-jurgen-jansma** [comment]
@brusch thank you for your review. I have used the `$inAdmin` argument instead of `$this->editmode` now. I have also added a thumbnail for Vimeo and Dailymotion as well.

File: models/Document/Editable/Video.php [hunk 1]
```diff
@@ -615,7 +615,7 @@ private function parseYoutubeId()
     /**
      * @return string
      */
-    private function getYoutubeCode()
+    private function getYoutubeCode($inAdmin = false)
```
Comment (blankse) [inline models/Document/Editable/Video.php]:
```suggestion
    private function getYoutubeCode(bool $inAdmin = false): string
```

File: models/Document/Editable/Video.php [hunk 2]
```diff
@@ -710,7 +716,7 @@ private function getYoutubeCode()
     /**
      * @return string
      */
-    private function getVimeoCode()
+    private function getVimeoCode($inAdmin = false)
```
Comment (blankse) [inline models/Document/Editable/Video.php]:
```suggestion
    private function getVimeoCode(bool $inAdmin = false): string
```

File: models/Document/Editable/Video.php [hunk 3]
```diff
@@ -789,7 +801,7 @@ private function getVimeoCode()
     /**
      * @return string
      */
-    private function getDailymotionCode()
+    private function getDailymotionCode($inAdmin = false)
```
Comment (blankse) [inline models/Document/Editable/Video.php]:
```suggestion
    private function getDailymotionCode(bool $inAdmin = false): string
```

File: models/Document/Editable/Video.php [hunk 4]
```diff
@@ -789,7 +801,7 @@ private function getVimeoCode()
     /**
      * @return string
      */
```
Comment (blankse) [inline models/Document/Editable/Video.php]:
```suggestion
```

File: models/Document/Editable/Video.php [hunk 5]
```diff
@@ -710,7 +716,7 @@ private function getYoutubeCode()
...
```
Comment (blankse) [inline models/Document/Editable/Video.php]:
```suggestion
```

File: models/Document/Editable/Video.php [hunk 6]
```diff
@@ -615,7 +615,7 @@ private function parseYoutubeId()
...
```
Comment (blankse) [inline models/Document/Editable/Video.php]:
```suggestion
```

**Review by blankse — CHANGES_REQUESTED** [review]
Please use type hints for new params

**Comment by bitfactory-jurgen-jansma** [comment]
@blankse I didn't add the type hinting because it wasn't use yet, and I expected that it wasn't there for a reason. However, I have added the type hinting to all edited methods.

**Comment by brusch** [comment]
Failing Codeception tests are not related to this PR ... caused by #13171 (fix)

**Pull request merged**